include/net/nrf91_lwm2m_app.rst

```diff
@@ -0,0 +1,195 @@
+business logic may be implemented inside custom resources, which are either proprietary to this application
+or may be following some external specification, like IPSO objects. Either way, they extend the resource tree
+by defining own objects on top of what LwM2M already defines.
+
+Before defining a proprietary resource, it is a good practice to check whether similar functionality is
+already defined in `LwM2M Object and Resource Registry`_.
+If there is already such definition you might want to use that, instead of defining your own.
+
+Please note that Zephyr's LwM2M library has only a limited support for some IPSO objects, so extending beyound those,
+requires changes to the internal engine as well.
+
+For this example, we are going to demonstrate how do we define a resource that follow Generic Sensor definition from IPSO.
+First, we look for "Generic Sensor" from the `LwM2M Object and Resource Registry`_ list
+which would give us that the resource object ID would be `3300`. Within that list, we could open a `raw XML`_ file
+or open a bit more human readable view from `the LwM2M editor`_ to see what resources are defined under the object ID 3300.
+
+.. _LwM2M Object and Resource Registry: http://openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html
+.. _raw XML: https://raw.githubusercontent.com/OpenMobileAlliance/lwm2m-registry/prod/version_history/3300-1_0.xml
+.. _the LwM2M editor: http://devtoolkit.openmobilealliance.org/OEditor/LWMOView?url=https%3A%2F%2Fraw.githubusercontent.com%2FOpenMobileAlliance%2Flwm2m-registry%2Fprod%2Fversion_history%2F3300-1_0.xml
+
+.. list-table:: Resource Definitions
+   :header-rows: 1
+   :widths: auto
+
+   * - ID
+     - Name
+     - Operations
+     - Instances
+     - Mandatory
+     - Type
+     - Description
+   * - 5700
+     - Sensor Value
+     - R
+     - Single
+     - Mandatory
+     - Float
+     - Last or Current Measured Value from the Sensor.
+   * - 5701
+     - Sensor Units
+     - R
+     - Single
+     - Optional
+     - String
+     - Measurement Units Definition.
+   * - 5601
+     - Min Measured Value
+     - R
+     - Single
+     - Optional
+     - Float
+     - The minimum value measured by the sensor since power ON or reset.
+   * - 5602
+     - Max Measured Value
+     - R
+     - Single
+     - Optional
+     - Float
+     - The maximum value measured by the sensor since power ON or reset.
+
+From the table above, we can see that resource ID of our sensor value would be `5700`.
+For the purpose of the tutorial, we only demonstrate that one resource, which is mandatory.
+As the LwM2M uses resource paths in form of `object ID/instance/resource ID` and our object ID is 3300
+and of course this is our first instance. Therefore our full path for the sensor value would be
+`3300/0/5700`. You need to use this path in the LwM2M library API.
+
+First thing we do, is to have read-function for our data. It is out of scope for this tutorial
```

Comment:
I know it'd be hard to cover every possible use of the library, but I think it's worth mentioning that registering a read callback is optional and is recommended if we want to read the data directly from a sensor on each READ operation. 

Another common case for the readable resource is to modify their value on event, there's no need to register a read callback then. A good exaple for this is the Push Button object - whenever we get an event that the button was pressed/released we just update its value through the lwm2m_engine with `lwm2m_engine_set_bool()`. When a READ operation is issued by the server, the engine will get the button value directly from the object internal data instead of reaching the read callback.

Reply:
Thanks, this got added by Uma.